lib/mistral_ai_llm_controller_example/mistral_ai_llm_controller_page.dart

```diff
@@ -0,0 +1,212 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:mistral_ai_chat_example_app/mistral_ai_llm_controller_example/logger_dialog.dart';
+import 'package:mistral_ai_chat_example_app/mistral_ai_llm_controller_example/model.dart';
+import 'package:mistral_ai_chat_example_app/mistral_ai_llm_controller_example/prompt.dart';
+import 'package:mistral_ai_chat_example_app/mistral_ai_llm_controller_example/utils.dart';
+import 'package:mistral_ai_chat_example_app/mistral_ai_summary_example/mistral_client.dart';
+import 'package:mistralai_client_dart/mistralai_client_dart.dart';
+
+class MistralAiLlmControllerPage extends StatefulWidget {
+  const MistralAiLlmControllerPage({super.key});
+
+  @override
+  State<MistralAiLlmControllerPage> createState() =>
+      _MistralAiLlmControllerPageState();
+}
+
+class _MistralAiLlmControllerPageState
+    extends State<MistralAiLlmControllerPage> {
+  final TextEditingController commandInputController = TextEditingController();
+  ControllerSettings controllerSettings = ControllerSettings();
+  bool showLoading = false;
+  String errorMessage = '';
+  String logger = '';
+
+  Future<String> _getResponseFromAi(String command) async {
+    setState(() {
+      showLoading = true;
+      logger += createStartCommandLog(
+        command,
+        controllerSettings,
+      );
+    });
+
+    final response = await mistralAIClient.chat(
+      ChatParams(
+        model: 'mistral-medium',
+        messages: [
+          ChatMessage(
+            role: 'system',
+            content: controllerDescription(
+              controllerSettings,
+            ),
+          ),
+          const ChatMessage(role: 'system', content: controllerExample),
+          ChatMessage(role: 'user', content: command),
+        ],
+      ),
+    );
+
+    setState(() {
+      showLoading = false;
+      logger += createResponseLog(response.choices.last.message.content);
```

Comment:
I think you could create a function like `appendToLog` or just `log` and hide the `logger +=` inside.

Reply:
Good idea it will be more readable, resolved in [5eba314](https://github.com/nomtek/flutter_ai_examples/pull/6/commits/5eba3146cedae75c570dd94fbee6c8154574fe42)